runelite-client/src/main/java/net/runelite/client/plugins/maxhit/MaxHitCalc.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2018, Magic fTail
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.maxhit;
+
+import net.runelite.api.Client;
+import net.runelite.api.Skill;
+import net.runelite.api.widgets.Widget;
+
+public class MaxHitCalc extends MaxHitPlugin
+{
+	public int melee(Client client, Widget meleeStrength)
+	{
+		double equipmentMeleeStrength = Double.parseDouble(meleeStrength.getText().replace("Melee strength: ", ""));
+		int str = client.getRealSkillLevel(Skill.STRENGTH);
+		double effectiveStrength = Math.floor((Math.floor((str + meleeBoost) * meleePrayerBonus) + 11) * voidMeleeBonus);
+		double maxHit = Math.floor(0.5 + effectiveStrength * (equipmentMeleeStrength + 64) / 640);
+		return (int)maxHit;
+	}
+
+	public int trident(Client client, Widget magicDamage, int i)
```

Comment:
Will this work for both versions of trident and slayer/slayer(e)? I would probably also name this differently/make it more generic.

Reply:
Pretty much every magic weapon is unique and then only one people really use is trident, so I decided to just go with that. The only difference between Trident and toxic is a flat max hit increase, the int. I didn't even consider slayer helmets, I'll add that in a similar fashion to how DH works